### ISSUE TITLE
Issue with pon.binsize

### ISSUE THREAD

**Issue by hraeder41** [mskilab-org/dryclean#31]
Hello,

I am attempting to use Dryclean with the PON provided at the bottom of the Readme, however I am receiving the below error:

`(Let's dryclean the genomes!)

Loading PON...
PON loaded
Loading coverage
Loading PON a.k.a detergent
Error in if (tumor.binsize != pon.binsize & testing == FALSE) { :
  argument is of length zero
Calls: <Anonymous>
2: (function ()
   traceback(2))()
1: dryclean_object$clean(cov = opt$input, center = opt$center, cbs = opt$cbs,
       cnsignif = opt$cnsignif, mc.cores = opt$cores, verbose = TRUE,
       use.blacklist = opt$blacklist, blacklist_path = opt$blacklist_path,
       germline.filter = opt$germline.filter, field = opt$field,
       testing = opt$testing)`

When running dryclean_object$clean in R directly, it appears that my coverage file has a 1000bp bin size as expected, but the PON is returning NULL when pon.binsize is set. Do you have any insight into what may be causing this?

**Comment by sebastian-brylka** [comment]
please try with this pon: https://mskilab-pipeline.s3.amazonaws.com/dryclean/pon/hg19/fixed.detergent.rds

**Comment by hraeder41** [comment]
Thank you very much! It looks like I am getting past this check now, however I am now getting the below error:

`Loading PON...
PON loaded
Loading coverage
Loading PON a.k.a detergent
Let's begin, this is whole exome/genome
Median-centering the sample
Initializing wash cycle
Using the detergent provided to start washing
lambdas calculated
calculating A and B
calculating v and s
Error in if (e < 1e-06 || k > maxiter) { :
  missing value where TRUE/FALSE needed
Calls: <Anonymous> -> wash_cycle -> apg_project
4: (function ()
   traceback(2))()
3: apg_project(m.vec, U, lambda1, lambda2)
2: wash_cycle(m.vec = m.vec, L.burnin = L.burnin, S.burnin = S.burnin,
       r = r, U.hat = U.hat, V.hat = V.hat, sigma.hat = sigma.hat)
1: dryclean_object$clean(cov = opt$input, center = opt$center, cbs = opt$cbs,
       cnsignif = opt$cnsignif, mc.cores = opt$cores, verbose = TRUE,
       use.blacklist = opt$blacklist, blacklist_path = opt$blacklist_path,
       germline.filter = opt$germline.filter, field = opt$field,
       testing = opt$testing)`

Is this also related to the PON, or is it more likely an issue with my coverage input?

**Comment by sebastian-brylka** [comment]
It is most likely the coverage input issue, are you using a mask?

**Comment by hraeder41** [comment]
I am not, I am using the direct "cov.rds" output from FragCounter. Also, one thing to note is that this is whole-exome data rather than whole-genome. Would that be a potential cause of the issue?

**Comment by sebastian-brylka** [comment]
yes, that is probably the cause, can you try setting 'wgs' parameter to FALSE when you initialize the PON object?
